File: src/testWithSpringBoot_2_1/java/org/openrewrite/java/spring/boot2/MigrateRestTemplateBuilderTimeoutByIntTest.java
/*
 * Copyright 2024 the original author or authors.
 * <p>
 * Licensed under the Moderne Source Available License (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * https://docs.moderne.io/licensing/moderne-source-available-license
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.openrewrite.java.spring.boot2;

import org.junit.jupiter.api.Test;
import org.openrewrite.DocumentExample;
import org.openrewrite.java.JavaParser;
import org.openrewrite.test.RecipeSpec;
import org.openrewrite.test.RewriteTest;

import static org.openrewrite.java.Assertions.java;

class MigrateRestTemplateBuilderTimeoutByIntTest implements RewriteTest {

    @Override
    public void defaults(RecipeSpec spec) {
        spec.recipe(new MigrateRestTemplateBuilderTimeoutByInt())
          .parser(JavaParser.fromJavaVersion().classpath("spring-boot", "spring-web", "spring-core"));
    }

    @DocumentExample
    @Test
    void changeDeprecatedMethods() {
        //language=java
        rewriteRun(
          java(
            """
              import org.springframework.boot.web.client.RestTemplateBuilder;
              import org.springframework.web.client.RestTemplate;

              class Test {
                  RestTemplate template = new RestTemplateBuilder()
                          .setConnectTimeout(1)
                          .setReadTimeout(1)
                          .build();
              }
              """,
            """
              import org.springframework.boot.web.client.RestTemplateBuilder;
              import org.springframework.web.client.RestTemplate;
              
              import java.time.Duration;
              
              class Test {
                  RestTemplate template = new RestTemplateBuilder()
                          .setConnectTimeout(Duration.ofMillis(1))
                          .setReadTimeout(Duration.ofMillis(1))
                          .build();
              }
              """
          )
        );
    }

    @Test
    void doNotChangeCurrentApi() {
        //language=java
        rewriteRun(
          java(
            """
              import org.springframework.web.client.RestTemplate;
              import org.springframework.boot.web.client.RestTemplateBuilder;
              import java.time.Duration;
              class Test {
                  RestTemplate template = new RestTemplateBuilder()
                      .setConnectTimeout(Duration.ofMillis(1)).build();
              }
              """
          )
        );
    }
}

File: src/testWithSpringBoot_2_1/kotlin/org/openrewrite/java/spring/boot2/MigrateRestTemplateBuilderTimeoutByIntTest.kt
/*
 * Copyright 2021 the original author or authors.
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * https://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.openrewrite.java.spring.boot2

import org.junit.jupiter.api.Test
import org.openrewrite.Recipe
import org.openrewrite.java.JavaParser
import org.openrewrite.java.JavaRecipeTest

class MigrateRestTemplateBuilderTimeoutByIntTest : JavaRecipeTest {
    override val parser: JavaParser
        get() = JavaParser.fromJavaVersion()
            .classpath("spring-boot", "spring-web")
            .build()

    override val recipe: Recipe
        get() = MigrateRestTemplateBuilderTimeoutByInt()

    @Test
    fun doNotChangeCurrentApi() = assertUnchanged(
        before = """
            import org.springframework.web.client.RestTemplate;
            import org.springframework.boot.web.client.RestTemplateBuilder;
            import java.time.Duration;
            class Test {
                RestTemplate template = new RestTemplateBuilder()
                    .setConnectTimeout(Duration.ofMillis(1)).build();
            }
        """
    )

    @Test
    fun changeDeprecatedMethods() = assertChanged(
        before = """
            import org.springframework.boot.web.client.RestTemplateBuilder;
            import org.springframework.web.client.RestTemplate;

            class Test {
                RestTemplate template = new RestTemplateBuilder()
                        .setConnectTimeout(1)
                        .setReadTimeout(1)
                        .build();
            }
        """,
        after = """
            import org.springframework.boot.web.client.RestTemplateBuilder;
            import org.springframework.web.client.RestTemplate;
            
            import java.time.Duration;
            
            class Test {
                RestTemplate template = new RestTemplateBuilder()
                        .setConnectTimeout(Duration.ofMillis(1))
                        .setReadTimeout(Duration.ofMillis(1))
                        .build();
            }
        """
    )
}
